src/api.js

```diff
@@ -486,6 +487,18 @@ let ensureTaskGroup = async (ctx, taskId, taskDef, res) => {
     }
   });
 
+  // Now we walso add the task to the group size counters as well
+  await ctx.TaskGroupSize.create({
```

Comment:
I'm not sure this is the ideal place to do this...
Here is the current process:
 1) TaskGroupMember.create(...)
 2) putDeadlineMessage
 3) TaskGroupSize.create(...)
 4) Task.create(...)
 5) Track dependencies (if needed)
 6) Pulse taskDefined
 7) putPendingMessage (if needed) 
 8) Pulse taskPending (if needed)

This ordering isn't arbitrary, we handle failure modes here... When we do listTaskGroup, we query the `TaskGroupMember` table, but for every entry we lookup the `Task` table to ensure the task was in fact created. When handling deadline messages we ignore cases where the `Task` table entry is missing, because the node could have crashed in the process of making it...

I would suggest that we insert in the `TaskGroupSize` table after `putDeadlineMessage`... Probably before `Task.Create(...)`... definitely before we track dependencies or `putPendingMessage`... So step 3.
Then we patch deadline resolution logic to delete the `TaskGroupSize` even if there was no entry in the `Task` table.

Reasoning:
If `queue.createTask` fails and isn't retried... Then if it crashes before:
 * step 2: Nothing happens (TaskGroupMember refres to Task that doesn't exist, the list logic will filter it out)
 * step 3: deadline message refers Task and TaskGroupSize entries that doesn't exist, nothing happens.
 * step 4: taskGroup-resolved message is delayed until the task.deadline when the deadline message, sees that the Task entry isn't defined, but the TaskGroupSize is defined... So it deletes TaskGroupSize and sends the message.
 * step 5: task completed message and taskGroupId resolved messages are delayed until task.deadline. Because even though the task was created, dependencies aren't tracking it and pending message wasn't sent... But deadline message will resolve it.
....

Notice that this way we always end up in a sane state...  Sometimes we have tasks (and task-groups) that weren't created successfully, and retries weren't followed up. The result would be that such tasks and task-groups are eventually resolved by deadline expiration. They simplify fail to become pending (and doesn't get picked up by a worker, even if they were initially pending).